openmls/src/tree/tests_and_kats/unit_tests/test_tree_truncation.rs

```diff
@@ -0,0 +1,146 @@
+use crate::{
+    ciphersuite::Ciphersuite,
+    credentials::{CredentialBundle, CredentialType},
+    group::{HandshakeMessageFormat, ManagedGroupCallbacks, ManagedGroupConfig, UpdatePolicy},
+    node::{Node, NodeType},
+    prelude::{KeyPackageBundle, LeafIndex},
+    test_utils::test_framework::{ActionType, ManagedTestSetup},
+    tree::RatchetTree,
+};
+
+#[test]
+fn test_trim() {
+    // Build a list of nodes, for which we need credentials and key package bundles
+    let mut nodes = vec![];
+    let mut key_package_bundles = vec![];
+    let ciphersuite = Ciphersuite::default();
+    for i in 0..10 {
+        let credential_bundle = CredentialBundle::new(
+            vec![i as u8],
+            CredentialType::Basic,
+            ciphersuite.signature_scheme(),
+        )
+        .unwrap();
+        let key_package_bundle =
+            KeyPackageBundle::new(&[ciphersuite.name()], &credential_bundle, vec![]).unwrap();
+
+        // We build a leaf node from the key packages
+        let leaf_node = Node {
+            node_type: NodeType::Leaf,
+            key_package: Some(key_package_bundle.key_package().clone()),
+            node: None,
+        };
+        key_package_bundles.push(key_package_bundle);
+        nodes.push(Some(leaf_node));
+        // We skip the last parent node (trees should always end with a leaf node)
+        if i != 9 {
```

Comment:
It's only a test. But this is dangerous. If the range of the loop is change, this breaks.

Reply:
Instead of hard-coding values for the tree size etc. It would be better to have it all variable and then also run this for a bunch of tree sizes. It's probably not necessary to cover all of `trim_tree` but it makes sure the test doesn't easily break.